src/script/proof.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <script/proof.h>
+
+const std::string strMessageMagic = "Bitcoin Signed Message:\n";
+
+namespace proof
+{
+
+Result SignMessage::Prepare(const std::vector<SignMessage>& entries, const std::string& message, std::set<CScript>& inputs_out, uint256& sighash_out, CScript& spk_out) const {
+    Result rv = Purpose::Prepare(m_scriptpubkey, inputs_out);
+    if (rv != RESULT_VALID) return rv;
+    CHashWriter hw(SER_DISK, 0);
+    std::string s = strMessageMagic + message;
+    hw << m_scriptpubkey << LimitedString<65536>(s);
```

Comment:
Is this intended to limit the string? I think it won't.

Reply:
Limiting was not the intention, really, and I'm not sure there is an attack vector for very large messages since these are passed between users outside of the proofs.

Reply:
What I wanted to say is that this constructor does not take into account the limit of 65536. The limit is only relevant in the `Unserialize` method, see https://github.com/bitcoin/bitcoin/blob/master/src/serialize.h#L503

Reply:
Oh, I see what you're saying. I can just change it to `s`. I used the limited string thing simply because I didn't realize strings were allowed as is.